controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -658,3 +692,13 @@ func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rookCephv1.CephClient{}).
 		Complete(r)
 }
+
+func generateHashForCephClient(storageConsumerName, claimID, cephUserType string) string {
+	cephClientName := &cephClientFormat{StorageConsumerName: storageConsumerName, ClaimID: claimID, CephUserType: cephUserType}
+	cephClient, err := json.Marshal(cephClientName)
+	if err != nil {
+		panic("failed to marshal")
```

Comment:
return error instead of panic and also return the actual error

Reply:
added log but keeping panic...since in other PR it was suggested to do panic when marshalling fails by @nb-ohad (if I'm not wrong)so keeping it, let me know if we want to remove it.

Reply:
Panic is ok here as we are not trying to marshal a user-given input and as such, we should not fail.